ALICE3/Tasks/pidRICHqa.cxx

```diff
@@ -120,26 +129,57 @@ struct richPidQaMc {
     }
 
     const AxisSpec sigAxis{1000, 0, 0.3, "Cherenkov angle (rad)"};
+    const AxisSpec lengthAxis{1000, 0, 3000, "Track length (cm)"};
+    const AxisSpec sigErrAxis{100, 0, 1, "Cherenkov angle error (rad)"};
     const AxisSpec nsigmaAxis{nBinsNsigma, minNsigma, maxNsigma, Form("N_{#sigma}^{RICH}(%s)", pT[pid_type])};
     const AxisSpec deltaAxis{nBinsDelta, minDelta, maxDelta, Form("#Delta(%s) (rad)", pT[pid_type])};
+    const AxisSpec etaAxis{100, -4, 4, "#it{#eta}"};
 
     histos.add("event/vertexz", ";Vtx_{z} (cm);Entries", kTH1F, {{100, -20, 20}});
-    histos.add("p/Unselected", "Unselected", kTH1F, {momAxis});
-    histos.add("p/Prim", "Primaries", kTH1F, {momAxis});
-    histos.add("p/Sec", "Secondaries", kTH1F, {momAxis});
-    histos.add("pt/Unselected", "Unselected", kTH1F, {momAxis});
-    histos.add("qa/signal", "", kTH1F, {sigAxis});
-    histos.add("qa/eta", ";#it{#eta}", kTH1F, {{100, -4, 4}});
-    histos.add("qa/signalerror", ";Cherenkov angle error (rad)", kTH1F, {{100, 0, 1}});
-    histos.add("qa/signalvsP", "", kTH2F, {momAxis, sigAxis});
-    histos.add("qa/signalvsPPrim", "", kTH2F, {momAxis, sigAxis});
-    histos.add("qa/signalvsPSec", "", kTH2F, {momAxis, sigAxis});
-    histos.add(hdelta[pid_type].data(), "", kTH2F, {momAxis, deltaAxis});
-    histos.add(hnsigma[pid_type].data(), "", HistType::kTH2F, {ptAxis, nsigmaAxis});
-    histos.add(hnsigmaprm[pid_type].data(), "Primary", HistType::kTH2F, {ptAxis, nsigmaAxis});
-    histos.add(hnsigmasec[pid_type].data(), "Secondary", HistType::kTH2F, {ptAxis, nsigmaAxis});
-
-    histos.add(hfrichnsigma[pid_type].data(), "", HistType::kTH2F, {ptAxis, nsigmaAxis});
+    histos.add("particle/p", "", kTH1F, {momAxis});
+    histos.add("particle/pt", "", kTH1F, {ptAxis});
+    histos.add("particle/eta", "", kTH1F, {etaAxis});
+    histos.add("tracks/p", "", kTH1F, {momAxis});
+    histos.add("tracks/pt", "", kTH1F, {ptAxis});
+    histos.add("tracks/eta", "", kTH1F, {etaAxis});
+    histos.add("tracks/length", "", kTH1F, {lengthAxis});
+
+#define MakeRICHHistos(rich)                                                                     \
+  histos.add(rich "/TrackSelection", "", kTH1F, {{10, 0.5, 10.5}});                              \
+  histos.get<TH1>(HIST(rich "/TrackSelection"))->GetXaxis()->SetBinLabel(1, "Tracks read");      \
+  histos.get<TH1>(HIST(rich "/TrackSelection"))->GetXaxis()->SetBinLabel(2, "Passed RICH");      \
+  histos.get<TH1>(HIST(rich "/TrackSelection"))->GetXaxis()->SetBinLabel(3, "Passed TOF");       \
+  histos.get<TH1>(HIST(rich "/TrackSelection"))->GetXaxis()->SetBinLabel(4, "Passed minLength"); \
+  histos.get<TH1>(HIST(rich "/TrackSelection"))->GetXaxis()->SetBinLabel(5, "Passed maxLength"); \
+  histos.get<TH1>(HIST(rich "/TrackSelection"))->GetXaxis()->SetBinLabel(6, "Passed eta");       \
+  histos.get<TH1>(HIST(rich "/TrackSelection"))->GetXaxis()->SetBinLabel(7, "Passed PDG");       \
```

Comment:
fyi 
the add() functions of the histogram registry now also return a pointer to the respective histogram
so one could simplify this part a bit (in case you find it more readable) by using
```
{
  auto trackSelHist = std::get<std::shared_ptr<TH1>>(histos.add(rich "/TrackSelection", "", kTH1F, {{10, 0.5, 10.5}}));
  trackSelHist->GetXaxis()->...
  ...
}
```

Reply:
Thanks Mario, indeed thanks a lot, however I personally don't find it particularly more elegant as I still have to cast it to TH1. I might consider it for the future, thanks a lot!